### PR TITLE
Provide a stub platform view embedder for tester to avoid emitting errors

### PR DESCRIPTION
**Pull request by dnfield**
Now that the tester provides a surface, tests involving platform views make it further along the rasterization path and end up emitting errors to the console about not supporting a platform view embedder. This provides a stub platform view embedder, and updates the test harness to check for messages containing `[ERROR` when running dart tests.

### CODE REVIEW

File: testing/run_tests.py [hunk 1]
```diff
@@ -419,16 +429,16 @@ def RunDartTests(build_dir, filter, verbose_dart_snapshot):
...
-    for dart_test_file in dart_observatory_tests:
-      if filter is not None and os.path.basename(dart_test_file) not in filter:
-        print("Skipping %s due to filter." % dart_test_file)
-      else:
-        print("Testing dart file %s with observatory enabled" % dart_test_file)
-        RunDartTest(build_dir, dart_test_file, verbose_dart_snapshot, True, True)
-        RunDartTest(build_dir, dart_test_file, verbose_dart_snapshot, False, True)
+  dart_tests = glob.glob('%s/*platform_view_test.dart' % dart_tests_dir)
+
+  # if 'release' not in build_dir:
```
Comment (chinmaygarde) [inline testing/run_tests.py:434]: Did you mean to check this in?

File: testing/run_tests.py [hunk 2]
```diff
@@ -247,7 +257,7 @@ def RunDartTest(build_dir, dart_file, verbose_dart_snapshot, multithreaded, enab
     threading = 'single-threaded'
 
   print("Running test '%s' using 'flutter_tester' (%s)" % (kernel_file_name, threading))
-  RunEngineExecutable(build_dir, 'flutter_tester', None, command_args)
+  RunEngineExecutable(build_dir, 'flutter_tester', None, command_args, forbidden_output=['[ERROR'])
```
Comment (chinmaygarde) [inline testing/run_tests.py:260]: What other errors are emitted? We should just patch the tester to provide necessary stubs.

**Review by chinmaygarde — COMMENTED** [review]
I am not sure we should suppress errors (and there is a bunch of python code commented out). Can we just provide the necessary stubs? Rest lgtm.

File: testing/run_tests.py (hunk 1 above)
Comment (dnfield) [reply testing/run_tests.py:434]: No, will fix

File: testing/run_tests.py (hunk 2 above)
Comment (dnfield) [reply testing/run_tests.py:260]: This is asserting the output doesn't contain an error log from the shell

**Comment by dnfield** [comment]
This is failing because we have two tests that are intended to fail, and they cause FML_LOG(ERROR)s to be printed.

I'm not sure if it would be better to remove the FML_LOG(ERROR), or to make those tests opt-out of allowing `[ERROR` to appear in the stdout. @chinmaygarde wdyt?

**Comment by dnfield** [comment]
I'm slightly inclined to remove the logs, since we have tests showing that we properly propagate a meaningful error message back to the Dart userland.

**Comment by dnfield** [comment]
I changed the logs to DLOG, and am allowing `unopt` builds to emit log errors.

**Comment by dnfield** [comment]
I belive the formatting issues are all fixed now - @chinmaygarde can you take a look again?

**Review by chinmaygarde — APPROVED** [review]

**Pull request merged**